core/tests/protractor_desktop/editorFeatures.js

```diff
@@ -973,6 +974,69 @@ describe('Exploration translation', function() {
     explorationEditorTranslationTab.changeTranslationLanguage('Hindi');
   });
 
+  it('should maintain its active sub-tab on saving draft and publishing changes'
+    , function() {
+      users.createUser('user@translationSubTab.com', 'userTranslationSubTab');
+      users.login('user@translationSubTab.com');
+      workflow.createExploration();
+
+      explorationEditorPage.navigateToSettingsTab();
+      explorationEditorSettingsTab.setTitle('Check');
+      explorationEditorSettingsTab.setCategory('Algorithms');
+      explorationEditorSettingsTab.setObjective('To check the translation tab');
+      explorationEditorPage.navigateToMainTab();
+      explorationEditorMainTab.setStateName('one');
+      explorationEditorMainTab.setContent(forms.toRichText(
+        'This is first card.'));
+      explorationEditorMainTab.setInteraction('NumericInput');
+      explorationEditorMainTab.addResponse(
+        'NumericInput', forms.toRichText('This is feedback1.'),
+        'two', true, 'Equals', 6);
+      var responseEditor = explorationEditorMainTab.getResponseEditor(
+        'default');
+      responseEditor.setFeedback(forms.toRichText('This is default_outcome.'));
+      explorationEditorMainTab.addHint('This is hint1.');
+      explorationEditorMainTab.addHint('This is hint2.');
+      explorationEditorMainTab.addSolution('NumericInput', {
+        correctAnswer: 6,
+        explanation: 'This is solution.'
+      });
+      explorationEditorMainTab.moveToState('two');
+      explorationEditorMainTab.setContent(forms.toRichText(
+        'This is second card.'));
+      explorationEditorMainTab.setInteraction('NumericInput');
+      explorationEditorMainTab.addResponse(
+        'NumericInput', forms.toRichText('This is feedback1.'),
+        'final card', true, 'Equals', 7);
+      var responseEditor = explorationEditorMainTab.getResponseEditor(
```

Comment:
done

Reply:
@DubeySandeep Seems like declaring responseEditor again is important, otherwise the tests are failing.
Hence, reverting the changes.

Reply:
I don't understand, can you please check what's going on here? (It's important to understand what we have written, right?)

Reply:
@DubeySandeep I checked the code, and I found that it's necessary to declare the variable again, because every time, the default response tab changes. Kindly have a look at the lines 44 and 288 of `ExplorationEditorMainTab.js` for reference.
Also, I used it in my code without thinking much earlier because it was redeclared quite many times in the same test whenever responseEditor was used. Thanks. This made my understanding of the function more clear.

Reply:
I think there is some misunderstanding here, I was asking you "why we are declaring the same variable again? i.e, `responseEditor` is declared and defined in line 995. so let's remove the var keyword."

Reply:
Oh, sorry for the misunderstanding. Your language was clear the first time itself. I mistook declaration for assignment. My bad.
Making the change.

Reply:
done.